bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/GarbageCollectorThread.java

```diff
@@ -183,6 +233,8 @@ public GarbageCollectorThread(ServerConfiguration conf,
                + majorCompactionThreshold + ", interval=" + majorCompactionInterval);
 
         lastMinorCompactionTime = lastMajorCompactionTime = MathUtils.now();
+        calendar = Calendar.getInstance();
```

Comment:
You have to create a new calendar at every execution.

You have to explicitly configure a timezone for this calendar, otherwise the configuration is not deterministic

Can you consider using java time API and not legacy Calendar?

Reply:
I think that using the default system timezone is ok, the user configure the specified time  period using their default system timezone usually.
What does you mean that explicitly configure a timezone?  Adding a new field to timezone in config file?